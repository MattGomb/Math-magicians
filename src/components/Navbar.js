import React from 'react';
import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <header className="navbar-container">
      <h1 className="navbar-title" data-testid="title">Math Magicians</h1>
      <ul className="navbar" data-testid="nav-links">
        <li className="nav-item" data-testid=""><Link to="/home">Home</Link></li>
        <li className="nav-item"><Link to="calculator">Calculator</Link></li>
        <li className="nav-item"><Link to="quotes">Quote</Link></li>
      </ul>
    </header>
  );
}

export default Navbar;
