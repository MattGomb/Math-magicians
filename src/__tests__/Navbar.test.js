import Navbar from '../components/Navbar';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('the calculator snapshot and react testing library', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><Navbar/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
