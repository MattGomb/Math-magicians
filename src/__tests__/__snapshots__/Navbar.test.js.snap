// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the calculator snapshot and react testing library renders correctly 1`] = `
<header
  className="navbar-container"
>
  <h1
    className="navbar-title"
  >
    Math Magicians
  </h1>
  <ul
    className="navbar"
  >
    <li
      className="nav-item"
    >
      <a
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        href="/quotes"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</header>
`;
